src/Core/Models/StaticStore/Plan.cs

```diff
@@ -28,7 +28,7 @@ public abstract record Plan
     public bool HasCustomPermissions { get; protected init; }
     public int UpgradeSortOrder { get; protected init; }
     public int DisplaySortOrder { get; protected init; }
-    public int? LegacyYear { get; protected init; }
+    public int? LegacyYear { get; set; }
```

Comment:
Yes, normally we would want to only set `LegacyYear` at instantiation. In this case, we need to update the value later depending on if the feature flag is enabled or not. If it's disabled, we need to display the old plans to the user instead of the new plans, and since `LegacyYear` determines which plans are displayed, it has to be null in that case.